# Bioinformatics-Sequence-Alignment
Pairwise sequence alignment of amino acid (protein) sequences via dynamic programming algorithms, local (Smith-Waterman) &amp; global (Needleman-Wunch)

Input

The input to your implementation should be a text file containing two sequences of amino acids, written on separate lines.Script accepts file as a command line argument, along with the following additional arguments:

alignment algorithm: local (Smith-Waterman) or global (Needleman-Wunsch)
scoring matrix (a square matrix in a specific format)
gap opening penalty (a negative integer)
gap extension penalty (a negative integer)





